design/status-design-update.md

```diff
@@ -0,0 +1,1166 @@
+
+- Inclusion: We need to have a clear set of rules about what classes of errors show up at the *includer* level and what at the *includee* level.
+  - We also don't put limits on how big this structure should be, so we need to ensure that the status block can provide useful information for large levels of complexity.
+- Composability: Because HTTPProxies can be included under more than one root, and can also be included in multiple places under a single root, Status needs to be able to work in both those cases.
+
+We also need to consider the interaction between service admins when one side changes something that breaks something for the other side.
+
+In addition, we need to keep some rules in mind about making the status useful:
+
+- the status must be able to indicate multiple errors, not just one.
+- errors must be able to be directly attributed to the part of the config that produced them.
+  - To help with this, the status should have a way to show errors with routes and includes separately.
+  - Most portions of the status must be namespaced by at least the fqdn of the root, if not some rendering of the include path.
+- the status must be able to indicate problems that will stop the HTTPProxy working as expected, but that are not syntactic errors with the HTTPProxy construct. For example, the referenced service is invalid.
+- the status should have a way to see the object as rendered by contour (that is, with merged conditions). This allows users to inspect what output Contour produces for a given input without using debug inputs or inspecting Envoy.
+
+It's also worth noting that `status` does not have to *only* be status.
+The `status` field in a Kubernetes object is generally used for any *computed information* (as opposed to the declarative configuration in `spec`)
+
+Lastly, it's important to note that, now that HTTPProxy is GA, we can't change the existing status fields very much, if at all.
+
+### Goals
+
+Overall:
+
+- Provide a better user experience for understanding HTTPProxy usage.
+
+Required:
+
+- Detail expected personas for users
+- Provide a mechanism for detailed information about invalid inclusions, including specifying what is added to each object, includer and included.
+- Provide a mechanism for detailed information about invalid routes/services.
+- Provide a mechanism for HTTPProxy administrators to see what conditions are added at each level of inclusion.
+- Provide the information required for higher-level parsing of the status (to enable both kubectl and other UI inspection).
+- An external tool must be able to build a representation of the HTTPProxy include DAG without speaking using only the Kubernetes API.
+- Provide a set of examples that can be used both for understanding this proposal and as the basis for a test suite for status.
+
+Optional:
+
+- Add usable printer columns for kubectl use (Fixes #1567)
+
+### Non-Goals
+
+- Considering how this will be implemented at anthing but the most basic level.
```

Comment:
It doesn't seem like a non-goal. I think the external tool building a DAG should be a non-goal.

Reply:
I meant of this initial proposal. Like I said just after, I'm trying to start from what users would want, and work from there, rather than starting with how it's implemented now and figuring out how to mutate it.